OpenRA.Mods.Common/Graphics/RectangleAnnotationRenderable.cs

```diff
@@ -0,0 +1,63 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Primitives;
+
+namespace OpenRA.Mods.Common.Graphics
+{
+	public struct RectangleAnnotationRenderable : IRenderable, IFinalizedRenderable
+	{
+		readonly WPos[] vertices;
+		readonly WPos effectivePos;
+		readonly int width;
+		readonly Color color;
+
+		public RectangleAnnotationRenderable(WPos diagVertice1, WPos diagVertice2, WPos effectivePos, int width, Color color)
+		{
+			WPos[] wposArray = { diagVertice1, new WPos(diagVertice2.X, diagVertice1.Y, 0), diagVertice2, new WPos(diagVertice1.X, diagVertice2.Y, 0) };
+
+			vertices = wposArray;
```

Comment:
Because of
![note 2019-12-06 164116](https://user-images.githubusercontent.com/13763394/70308999-77361880-1847-11ea-9be1-0021e7e1b30e.jpg)
I just don't know why it keeps telling me that I must replace `,` into `;`. If I follow it's orders, VS tell me this is nonsense in syntax either. Finally, I got this right by doing this. Do you have any good idea on this?

In addition, my VS C# tell me `this` in `this.vertices = { diagVertice1, new WPos(diagVertice2.X, diagVertice1.Y, 0), diagVertice2, new WPos(diagVertice1.X, diagVertice2.Y, 0) }` is unnecessary, and fail to compile everytime on github becasue 'unnecessary'. I feel very weird about it, but also have to do as it "debugs".

Reply:
Ah sorry, I derped. Try
`vertices = new[] { diagVertice1, new WPos(diagVertice2.X, diagVertice1.Y, 0), diagVertice2, new WPos(diagVertice1.X, diagVertice2.Y, 0) };`

Reply:
Thanks, that works good in VS.